tests/conftest.py

```diff
@@ -388,6 +389,33 @@ def get_specified_duts(request):
     return duts
 
 
+def add_custom_msg(request, key, val):
+    """
+    Add a custom message to the cache with a specified prefix.
+
+    At the end of each test, cached custom messages will be logged and collected
+    by Kusto for debugging purposes.
+
+    Args:
+        request: The pytest request object.
+        key (str): The key for the custom message. Use '.' to separate different
+                   levels of keys (e.g., "foo.bar.baz" will be stored as
+                   { "foo": { "bar": { "baz": val } } }).
+        val: The value to be stored in the cache under the specified key.
+    """
+    request.config.cache.set(f"{CUSTOM_MSG_PREFIX}.{key}", val)
+
+
+def pytest_sessionstart(session):
```

Comment:
Have you tested the pytest_sessionstart is behaving as expected?

Reply:
Yes, confirmed it is the very first thing to be called before the test.

Reply:
But I noticed they remove all the cache files before each session. So I don't think this sessionstart will delete any cache in real scanario